Naughts_and_Crosses.py

```diff
@@ -1,122 +1,237 @@
-                y = ()
-        return y
-
-    def turn(self):
-        x_row = choice_row()
-        y_column = choice_column()
-        if board_state[x_row][y_column] == 'X' or board_state[x_row][y_column] == 'O':
-            print('Square\'s already full idiot') 
-            turn(self)
+            print("That's not a player number...")
+            self.first_turn()
+
+    def turn(self): #Checks the current player, makes them do a turn based on what type of player they are then
+        if self.current_player.player_type == 'H':  #takes the slot they used out of the empty spaces list
+            choices = self.current_player.human_choice()
+            if choices in self.board.empty_spaces:
+                self.board.board_state[choices[0]][choices[1]] = self.current_player.symbol
+                self.board.empty_spaces.remove(choices)
+
+            else:
+                print('That space is not available douche. Choose again')
+                self.turn()
         else:
-            board_state[x_row][y_column] = self.symbol
-            return board_state
-
-class AI():
-	def __init__(self, symbol):
-		self.symbol = 'O'
-    def turn(self):
-        x_row = randrange(3)
-        y_column = randrange(3)
-        if board_state[x_row][y_column] == 'X' or board_state[x_row][y_column] == 'O':
-            turn(self)
+            choices = self.current_player.ai_choice()
+            if choices in self.board.empty_spaces:
+                self.board.board_state[choices[0]][choices[1]] = self.current_player.symbol
+                self.board.empty_spaces.remove(choices)
+            else:
+                self.turn()
+
+    def change_player(self): #Checks which players turn it is and then switches it then tells the next person who's go
+        if self.current_player == self.p1: # it is
+            self.current_player = self.p2
         else:
-            board_state[x_row][y_column] = self.symbol
-            return board_state
-
-
-
-  # Start the game here with asking the name and then getting into the start game function #
-  #Commit the Player1 turn function #
-  #Check win condition #
-  #Have AI or Player2 do their turn #
-  #Check win condition #
-  #Repeat until win condition is in there or board is full #
-  #Ask if they would like to play again, then ask if they would like to player against a second player or AI again #
-  
-
-
-
-
+            self.current_player = self.p1
+        print('It is now ' + self.current_player.name + "'s go")
+
+    def check_draw(self): #Checks the empty spaces list and if there is no empty spaces, it's a draw
+        if not self.board.check_free_spaces():
+            self.running = False
+            print("Well great. That's a draw.")
+            return True
+
+    def check_win(self): #Checks the board to see if there is a player symbol in all the spaces of each win condition
+        for i in range((self.board.board_size * 2) + 2):
+                if all(
```

Comment:
You're over indented here, but also:
``` python
for j in range(self.board.board_size):
    point = self.board.win_condition[i][j]
    if all(self.board.board_state[point[0]][point[1]] == self.current_player.symbol):
        blah
```
Using one-line for loops, though nice, are not always the way forward. And always redefine complex variables to simplify them.